src/main/java/homework_3/Engine.java

```diff
@@ -0,0 +1,60 @@
+package homework_3;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/*
+The class must be declared as final.
+Data members in the class must be declared as private & final.
+A parameterized constructor should initialize all the fields.
+Deep Copy of objects should be performed in the getter methods.
+No setters.
+ */
+
+public final class Engine {
```

Comment:
This class is mutable. 
![image](https://user-images.githubusercontent.com/58872509/134060118-addac9f5-a6d9-49fd-b9b4-3a5c65f8e827.png)